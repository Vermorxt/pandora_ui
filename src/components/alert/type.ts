import { I_Ui_Sizes } from './../_types/sizes/sizes'
import { I_Variants } from '../_types/colors/ui_variants'
import { HTMLProps, MouseEventHandler } from 'react'

export interface Ui_AlertProps extends I_Variants, Omit<HTMLProps<HTMLDivElement>, 'size'> {
  size?: I_Ui_Sizes
  noIcon?: boolean
  onDenyButtonText?: string
  onAcceptButtonText?: string
  onDeny?: MouseEventHandler<HTMLButtonElement> | undefined
  onAccept?: MouseEventHandler<HTMLButtonElement> | undefined
}
