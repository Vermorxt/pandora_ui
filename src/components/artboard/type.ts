import { T_Variants } from '../_types/colors/ui_variants'
import { HTMLProps } from 'react'

export interface Ui_ArtboardProps extends HTMLProps<HTMLSpanElement>, T_Variants {
  phone1?: boolean
  phone2?: boolean
  phone3?: boolean
  phone4?: boolean
  phone5?: boolean
  phone6?: boolean
  horizontal?: boolean
  demo?: boolean
}
