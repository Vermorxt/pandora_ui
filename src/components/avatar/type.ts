import { T_Variants } from 'components/_types/colors/ui_variants'
import { T_Ui_Mask } from 'components/_types/mask/mask'
import { HTMLProps } from 'react'
import { T_Ui_Size } from '../_types/sizes/sizes'

type Ui_BorderRadius = 'xl' | '2xl' | '3xl' | 'lg' | 'full'
type Ui_Ring = 'primary' | 'secondary' | boolean
type Ui_RingOffset = '0' | '1' | '2' | '4' | '8'
type Ui_RingOffsetBackground = T_Variants // NOTE: https://tailwindcss.com/docs/ring-offset-color

interface Ui_Avatar_ring {
  ring?: boolean
  ringPrimary?: boolean
  ringSecondary?: boolean
  ringOffset?: Ui_RingOffset
  ringOffsetBackground?: Ui_RingOffsetBackground
}

type Ui_Avatar_online = 'online' | 'offline' | null
interface Ui_Avatar_onlineStatus {
  onlineStatus?: Ui_Avatar_online
}

export interface Ui_AvatarProps extends Omit<HTMLProps<HTMLDivElement>, 'size'>, Ui_Avatar_onlineStatus {
  borderRadius?: Ui_BorderRadius
  size?: T_Ui_Size
  mask?: T_Ui_Mask
  ring?: Ui_Ring
  ringOffset?: Ui_RingOffset
  ringOffsetBackground?: Ui_RingOffsetBackground
  src?: string
  usePlaceHolder?: string // NOTE: content of last avatar item (e.g.: +99)
}
