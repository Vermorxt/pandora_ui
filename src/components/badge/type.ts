import { T_Ui_Size } from './../_types/sizes/sizes'
import { HTMLProps } from 'react'
import { I_Variants } from '../_types/colors/ui_variants'

export interface Ui_Badge_variants {
  outline?: boolean
}

export interface Ui_BadgeProps extends Omit<HTMLProps<HTMLSpanElement>, 'size'>, I_Variants, Ui_Badge_variants {
  size?: T_Ui_Size
}
