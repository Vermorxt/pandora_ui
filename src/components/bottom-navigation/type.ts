import { T_Variants } from 'components/_types/colors/ui_variants'
import { T_Ui_Size } from 'components/_types/sizes/sizes'
import { HTMLProps } from 'react'
import { I_Variants } from '../_types/colors/ui_variants'

export interface BottomNavigation {
  id?: string | number
  title?: string
  active?: boolean
  disabled?: boolean
  onClick?: (args?: any) => void
  icon?: { __html: string }
}

export interface Ui_BottomNavigationProps extends Omit<HTMLProps<HTMLDivElement>, 'size'>, I_Variants {
  size?: T_Ui_Size
  className?: string
  activeColor?: T_Variants
  navigation?: BottomNavigation[]
  bgColor?: T_Variants
  rounded?: boolean
  shadow?: 'sm' | 'lg' | 'xl' | '2xl' | 'md' | 'none' | boolean
}
