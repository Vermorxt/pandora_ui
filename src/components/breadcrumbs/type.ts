import { HTMLProps } from 'react'
import { T_Variants } from '../_types/colors/ui_variants'
import { I_Ui_Sizes } from '../_types/sizes/sizes'

export interface Ui_BreadcrumbsProps extends HTMLProps<HTMLSpanElement>, T_Variants, I_Ui_Sizes {}
