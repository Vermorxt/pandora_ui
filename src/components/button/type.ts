import { T_Ui_Size } from './../_types/sizes/sizes'
import { HTMLProps } from 'react'
import { I_Variants } from '../_types/colors/ui_variants'
import { I_Ui_Sizes } from '../_types/sizes/sizes'
import { TooltipProps } from '../_types/tooltip/tooltip-props'

export interface Ui_Button_variants extends I_Variants {
  active?: boolean
  disabled?: boolean
  glass?: boolean
  circle?: boolean
}

export interface Ui_Button_singleClass {
  loading?: boolean
  noAnimation?: boolean
}

export type Ui_ButtonType = 'button' | 'submit' | 'reset' | undefined

export interface Ui_ButtonProps
  extends Omit<HTMLProps<HTMLButtonElement>, 'size'>,
    Ui_Button_variants,
    Ui_Button_singleClass {
  size: T_Ui_Size | 'block' | 'wide' | 'full'
  outline?: boolean
  tooltip?: TooltipProps
  type?: Ui_ButtonType
}
