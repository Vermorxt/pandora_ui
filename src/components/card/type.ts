import { HTMLProps } from 'react'
import { I_Variants, T_Variants } from '../_types/colors/ui_variants'

interface Ui_Card_style {
  variant?: T_Variants
  bgBase?: '100' | '200' | '300'
  size?: 'w-96' | string
  actionPlacement?: 'start' | 'center' | 'between'
  cardSide?: 'large' | boolean
}
export interface Ui_CardProps extends Omit<HTMLProps<HTMLDivElement>, 'size'>, Ui_Card_style {
  imageAsBackground?: boolean
  compact?: boolean
  itemsCenter?: boolean
  textCenter?: boolean
  glass?: boolean
}
