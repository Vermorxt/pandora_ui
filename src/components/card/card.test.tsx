import '@testing-library/jest-dom'
import { render } from '@testing-library/react'
import { checkAccessibility, itSupportsSystemProps } from '../../_tests-utils'
import { itHasCssClass } from '../../_tests-utils/it-has-color-css-class'
import { Ui_CardProps } from './type'
import Ui_Card from './Card'
import React from 'react'
const defaultProps: Ui_CardProps = {}

const selector = 'card'

describe('@cms/core/Ui_Card', () => {
  checkAccessibility([
    <Ui_Card {...defaultProps} ref={null}>
      Test
    </Ui_Card>,
  ])
  itSupportsSystemProps({
    component: Ui_Card,
    props: defaultProps,
    // refType: HTMLDivElement,
  })

  // NOTE: (possible debug): debugComponent(<Ui_Card {...defaultProps} ref={null} />)

  it('should match snapshot', () => {
    const { container } = render(<Ui_Card {...defaultProps} ref={null} />)
    expect(container).toMatchSnapshot()
  })

  it('renders IMAGE_AS_BACKGROUND status based on prop', () => {
    const { container: available } = render(<Ui_Card {...defaultProps} imageAsBackground ref={null} />)
    const { container: notAvailable } = render(<Ui_Card {...defaultProps} ref={null} />)

    expect(available.querySelectorAll(`.image-full`)).toHaveLength(1)
    expect(notAvailable.querySelector(`.image-full`)).toBeNull()
  })

  it('renders W-96 status based on prop', () => {
    const { container: available } = render(<Ui_Card {...defaultProps} size="96" ref={null} />)
    const { container: notAvailable } = render(<Ui_Card {...defaultProps} ref={null} />)

    expect(available.querySelectorAll(`.w-96`)).toHaveLength(1)
    expect(notAvailable.querySelectorAll(`.w-96`)).toHaveLength(1)
  })

  it('renders COMPACT status based on prop', () => {
    const { container: available } = render(<Ui_Card {...defaultProps} compact ref={null} />)
    const { container: notAvailable } = render(<Ui_Card {...defaultProps} ref={null} />)

    expect(available.querySelectorAll(`.card-compact`)).toHaveLength(1)
    expect(notAvailable.querySelector(`.card-compact`)).toBeNull()
  })

  it('renders COLOR classes', async () => {
    const bgBase100 = '100'
    const bgBase200 = '200'
    const bgBase300 = '300'

    await itHasCssClass(<Ui_Card variant="primary" />, `.bg-primary`)
    await itHasCssClass(<Ui_Card variant="secondary" />, `.bg-secondary`)
    await itHasCssClass(<Ui_Card variant="neutral" />, `.bg-neutral`)
    await itHasCssClass(<Ui_Card variant="primary" />, `.text-primary-content`)
    await itHasCssClass(<Ui_Card variant="secondary" />, `.text-secondary-content`)
    await itHasCssClass(<Ui_Card variant="neutral" />, `.text-neutral-content`)
    await itHasCssClass(<Ui_Card glass />, `.glass`)
    await itHasCssClass(<Ui_Card cardSide />, `.card-side`)
    await itHasCssClass(<Ui_Card bgBase={bgBase100} />, `.bg-base-${bgBase100}`)
    await itHasCssClass(<Ui_Card bgBase={bgBase200} />, `.bg-base-${bgBase200}`)
    await itHasCssClass(<Ui_Card bgBase={bgBase300} />, `.bg-base-${bgBase300}`)
  })
})
