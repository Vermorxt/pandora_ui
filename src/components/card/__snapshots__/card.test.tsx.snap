// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@cms/core/Ui_Card should match snapshot 1`] = `
<div>
  <div
    class="card shadow-xl style-override
      
      
      bg-base-100
      
      
      
      
      
      w-96
      "
  />
</div>
`;
